### PR TITLE
govc: add tasks and task.cancel commands

### PR DESCRIPTION
**Pull request by dougm**

### CODE REVIEW

File: govc/task/recent.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+			if info.StartTime != nil {
+				start = info.StartTime.Format(stamp)
+			}
+
+			var result string
+
+			if info.CompleteTime == nil {
+				result = fmt.Sprintf("%2d%% %s", info.Progress, info.Task)
+			} else {
```
Comment (caglar10ur) [inline govc/task/recent.go]: [minor] [feel free to ignore]

```
result = fmt.Sprintf("%2d%% %s", info.Progress, info.Task)
if info.CompleteTime != nil {
...
}

maybe?

File: govc/task/recent.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	if err != nil {
+		return nil
+	}
+
+	defer v.Destroy(context.Background())
+
+	v.Follow = cmd.follow
+
+	stamp := "15:04:05"
+	tmpl := "%-30s %-30s %13s %9s %9s %9s %s\n"
```
Comment (caglar10ur) [inline govc/task/recent.go]: [cosmetic] [feel free to ignore] Another possibility could be using https://golang.org/pkg/text/tabwriter/ :P

File: object/task.go [hunk 3]
```diff
@@ -51,3 +52,11 @@ func (t *Task) WaitForResult(ctx context.Context, s progress.Sinker) (*types.Tas
 	p := property.DefaultCollector(t.c)
 	return task.Wait(ctx, t.Reference(), p, s)
 }
+
+func (t *Task) Cancel(ctx context.Context) error {
```
Comment (caglar10ur) [inline object/task.go]: 👍

File: govc/task/recent.go (hunk 2 above)
Comment (dougm) [inline govc/task/recent.go]: Yes, I started with that, but it doesn't work out well when using the '-f' flag to follow, so I opted for fixed size format string.

**Comment by dougm** [comment]
Here's an example of how it looks, using vic-machine create + delete:

```console
% govc tasks -f
Task                           Target                             Initiator    Queued   Started Completed Result
CreateVM                       vm                                      root  16:36:35  16:36:35         -  0% Task:haTask-ha-folder-vm-vim.Folder.createVm-150161839
CreateVM                       vm                                      root  16:36:35  16:36:35  16:36:35 success (59.341ms)
ReconfigVM                     dougm-vch                               root  16:36:35  16:36:35  16:36:35 success (19.487ms)
ReconfigVM                     dougm-vch                               root  16:36:35  16:36:35         -  0% Task:haTask-418-vim.VirtualMachine.reconfigure-150161866
ReconfigVM                     dougm-vch                               root  16:36:35  16:36:35  16:36:35 success (37.851ms)
PowerOnVM                      dougm-vch                               root  16:36:50  16:36:50         -  0% Task:haTask-418-vim.VirtualMachine.powerOn-150161924
PowerOnVM                      dougm-vch                               root  16:36:50  16:36:50  16:36:52 success (1.662817s)
QueryVirtualDiskInfo           [datastore1] dougm-vch/VIC/56*          root  16:37:15  16:37:15  16:37:15 success (1.649ms)
DeleteVirtualDisk              [datastore1] dougm-vch/VIC/56*          root  16:40:02  16:40:02  16:40:02 success (8.535ms)
DeleteVirtualDisk              [datastore1] dougm-vch/VIC/56*          root  16:40:02  16:40:02  16:40:02 File /vmfs/volumes/5963da6b-c4f62903-2f96-000c29565af2/dougm-vch/VIC/564d8987-968a-31ff-101e-92e894a26ee2/images/scratch/scratch-flat.vmdk was not found
PowerOffVM                     dougm-vch                               root  16:40:02  16:40:02         -  0% Task:haTask-418-vim.VirtualMachine.powerOff-150162285
PowerOffVM                     dougm-vch                               root  16:40:02  16:40:02  16:40:02 success (172.183ms)
ReconfigVM                     dougm-vch                               root  16:40:02  16:40:02         -  0% Task:haTask-418-vim.VirtualMachine.reconfigure-150162296
ReconfigVM                     dougm-vch                               root  16:40:02  16:40:02  16:40:02 success (46.634ms)
ReconfigVM                     dougm-vch                               root  16:40:02  16:40:02         -  0% Task:haTask-418-vim.VirtualMachine.reconfigure-150162310
ReconfigVM                     dougm-vch                               root  16:40:02  16:40:02  16:40:02 success (35.986ms)
VirtualMachine.Destroy         dougm-vch                               root  16:40:02  16:40:02         -  0% Task:haTask-418-vim.ManagedEntity.destroy-150162319
VirtualMachine.Destroy         dougm-vch                               root  16:40:02  16:40:02  16:40:02 success (65.98ms)
ResourcePool.Destroy           dougm-vch                               root  16:40:02  16:40:02  16:40:02 success (9.061ms)
```

**Pull request merged**